lib/request.js

```diff
@@ -78,6 +78,17 @@ function buildRequestWithTrustProxy (R, trustProxy) {
         }
         return this.headers.host || this.headers[':authority']
       }
+    },
+    protocol: {
+      get () {
+        if (this.headers['x-forwarded-proto']) {
```

Comment:
I think this should be a check on `trustProxy`? See https://github.com/fastify/fastify/issues/2577

Reply:
Not necessary. The calling method is called `buildRequestWithTrustProxy`. So the check for `trustProxy` was made before.